File: backend/models/claim.py
from datetime import datetime, timezone

from sqlalchemy import DateTime, Float, ForeignKey, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from ..core.db import Base


class Claim(Base):
    __tablename__ = "claims"
    id: Mapped[int] = mapped_column(primary_key=True)
    enterprise_id: Mapped[int] = mapped_column(ForeignKey("enterprises.id"))
    person_id: Mapped[int] = mapped_column(ForeignKey("insured_people.id"))
    claim_no: Mapped[str] = mapped_column(String(80), unique=True)
    description: Mapped[str] = mapped_column(Text, default="")
    status: Mapped[str] = mapped_column(String(30), default="reported")
    amount: Mapped[float] = mapped_column(Float, default=0)
    accident_at: Mapped[str] = mapped_column(String(30), default="")
    accident_place: Mapped[str] = mapped_column(String(200), default="")
    accident_type: Mapped[str] = mapped_column(String(60), default="工伤事故")
    hospital: Mapped[str] = mapped_column(String(160), default="")
    diagnosis: Mapped[str] = mapped_column(Text, default="")
    medical_cost: Mapped[float] = mapped_column(Float, default=0)
    contact_name: Mapped[str] = mapped_column(String(80), default="")
    contact_phone: Mapped[str] = mapped_column(String(30), default="")
    insurer_report_no: Mapped[str] = mapped_column(String(100), default="")
    current_handler: Mapped[str] = mapped_column(String(80), default="平台理赔专员")
    deadline: Mapped[str] = mapped_column(String(30), default="")
    approved_amount: Mapped[float] = mapped_column(Float, default=0)
    paid_at: Mapped[str] = mapped_column(String(30), default="")
    rejection_reason: Mapped[str] = mapped_column(Text, default="")
    review_note: Mapped[str] = mapped_column(Text, default="")
    sla_deadline: Mapped[str] = mapped_column(String(30), default="")
    risk_level: Mapped[str] = mapped_column(String(20), default="normal")
    created_at: Mapped[datetime] = mapped_column(DateTime, default=lambda: datetime.now(timezone.utc))


class ClaimTimeline(Base):
    __tablename__ = "claim_timelines"
    id: Mapped[int] = mapped_column(primary_key=True)
    claim_id: Mapped[int] = mapped_column(ForeignKey("claims.id"))
    node: Mapped[str] = mapped_column(String(40))
    action: Mapped[str] = mapped_column(String(100))
    note: Mapped[str] = mapped_column(Text, default="")
    operator: Mapped[str] = mapped_column(String(80), default="系统")
    created_at: Mapped[datetime] = mapped_column(DateTime, default=lambda: datetime.now(timezone.utc))


class ClaimDocument(Base):
    __tablename__ = "claim_documents"
    id: Mapped[int] = mapped_column(primary_key=True)
    claim_id: Mapped[int] = mapped_column(ForeignKey("claims.id"))
    name: Mapped[str] = mapped_column(String(160))
    url: Mapped[str] = mapped_column(Text, default="")
    doc_type: Mapped[str] = mapped_column(String(40), default="other")
    status: Mapped[str] = mapped_column(String(30), default="uploaded")
    review_note: Mapped[str] = mapped_column(Text, default="")
    created_at: Mapped[datetime] = mapped_column(DateTime, default=lambda: datetime.now(timezone.utc))

File: java-backend/src/main/java/com/xbb/baojing/claim/Claim.java
package com.xbb.baojing.claim;

import java.time.LocalDateTime;
import java.util.List;

public class Claim {
    private Integer id;
    private Integer enterpriseId;
    private Integer personId;
    private String claimNo;
    private String description = "";
    private String status = "reported";
    private double amount = 0;
    private String accidentAt = "";
    private String accidentPlace = "";
    private String accidentType = "工伤事故";
    private String hospital = "";
    private String diagnosis = "";
    private double medicalCost = 0;
    private String contactName = "";
    private String contactPhone = "";
    private String insurerReportNo = "";
    private String currentHandler = "平台理赔专员";
    private String deadline = "";
    private double approvedAmount = 0;
    private String paidAt = "";
    private String rejectionReason = "";
    private String reviewNote = "";
    private String slaDeadline = "";
    private String riskLevel = "normal";
    private LocalDateTime createdAt;

    // response-only, joined/computed — claim_payload() in services/claims.py
    private String enterpriseName;
    private String personName;
    private String idNumber;
    private String positionName;
    private String actualEmployerName;
    private String policyNo;
    private String planName;
    private String insurer;
    private int documentCount;
    private int missingCount;
    private List<String> missingTypes;
    private int completePercent;
    private Integer deadlineDays;
    private boolean deadlineOverdue;
    private boolean slaOverdue;
    private String calculatedRisk;

    public Integer getId() { return id; }
    public void setId(Integer id) { this.id = id; }
    public Integer getEnterpriseId() { return enterpriseId; }
    public void setEnterpriseId(Integer v) { this.enterpriseId = v; }
    public Integer getPersonId() { return personId; }
    public void setPersonId(Integer v) { this.personId = v; }
    public String getClaimNo() { return claimNo; }
    public void setClaimNo(String v) { this.claimNo = v; }
    public String getDescription() { return description; }
    public void setDescription(String v) { this.description = v; }
    public String getStatus() { return status; }
    public void setStatus(String v) { this.status = v; }
    public double getAmount() { return amount; }
    public void setAmount(double v) { this.amount = v; }
    public String getAccidentAt() { return accidentAt; }
    public void setAccidentAt(String v) { this.accidentAt = v; }
    public String getAccidentPlace() { return accidentPlace; }
    public void setAccidentPlace(String v) { this.accidentPlace = v; }
    public String getAccidentType() { return accidentType; }
    public void setAccidentType(String v) { this.accidentType = v; }
    public String getHospital() { return hospital; }
    public void setHospital(String v) { this.hospital = v; }
    public String getDiagnosis() { return diagnosis; }
    public void setDiagnosis(String v) { this.diagnosis = v; }
    public double getMedicalCost() { return medicalCost; }
    public void setMedicalCost(double v) { this.medicalCost = v; }
    public String getContactName() { return contactName; }
    public void setContactName(String v) { this.contactName = v; }
    public String getContactPhone() { return contactPhone; }
    public void setContactPhone(String v) { this.contactPhone = v; }
    public String getInsurerReportNo() { return insurerReportNo; }
    public void setInsurerReportNo(String v) { this.insurerReportNo = v; }
    public String getCurrentHandler() { return currentHandler; }
    public void setCurrentHandler(String v) { this.currentHandler = v; }
    public String getDeadline() { return deadline; }
    public void setDeadline(String v) { this.deadline = v; }
    public double getApprovedAmount() { return approvedAmount; }
    public void setApprovedAmount(double v) { this.approvedAmount = v; }
    public String getPaidAt() { return paidAt; }
    public void setPaidAt(String v) { this.paidAt = v; }
    public String getRejectionReason() { return rejectionReason; }
    public void setRejectionReason(String v) { this.rejectionReason = v; }
    public String getReviewNote() { return reviewNote; }
    public void setReviewNote(String v) { this.reviewNote = v; }
    public String getSlaDeadline() { return slaDeadline; }
    public void setSlaDeadline(String v) { this.slaDeadline = v; }
    public String getRiskLevel() { return riskLevel; }
    public void setRiskLevel(String v) { this.riskLevel = v; }
    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime v) { this.createdAt = v; }
    public String getEnterpriseName() { return enterpriseName; }
    public void setEnterpriseName(String v) { this.enterpriseName = v; }
    public String getPersonName() { return personName; }
    public void setPersonName(String v) { this.personName = v; }
    public String getIdNumber() { return idNumber; }
    public void setIdNumber(String v) { this.idNumber = v; }
    public String getPositionName() { return positionName; }
    public void setPositionName(String v) { this.positionName = v; }
    public String getActualEmployerName() { return actualEmployerName; }
    public void setActualEmployerName(String v) { this.actualEmployerName = v; }
    public String getPolicyNo() { return policyNo; }
    public void setPolicyNo(String v) { this.policyNo = v; }
    public String getPlanName() { return planName; }
    public void setPlanName(String v) { this.planName = v; }
    public String getInsurer() { return insurer; }
    public void setInsurer(String v) { this.insurer = v; }
    public int getDocumentCount() { return documentCount; }
    public void setDocumentCount(int v) { this.documentCount = v; }
    public int getMissingCount() { return missingCount; }
    public void setMissingCount(int v) { this.missingCount = v; }
    public List<String> getMissingTypes() { return missingTypes; }
    public void setMissingTypes(List<String> v) { this.missingTypes = v; }
    public int getCompletePercent() { return completePercent; }
    public void setCompletePercent(int v) { this.completePercent = v; }
    public Integer getDeadlineDays() { return deadlineDays; }
    public void setDeadlineDays(Integer v) { this.deadlineDays = v; }
    public boolean isDeadlineOverdue() { return deadlineOverdue; }
    public void setDeadlineOverdue(boolean v) { this.deadlineOverdue = v; }
    public boolean isSlaOverdue() { return slaOverdue; }
    public void setSlaOverdue(boolean v) { this.slaOverdue = v; }
    public String getCalculatedRisk() { return calculatedRisk; }
    public void setCalculatedRisk(String v) { this.calculatedRisk = v; }
}
